### DISCUSSION TITLE
Preference between client-side-auth or server-side

### DISCUSSION THREAD

**Discussion by quan612** [nextauthjs/next-auth#8217] [Help]
Hi,

I tried to search around but not seeing anyone having the same question. Just wondering if people prefer using server side auth through getServerSession(), or client side auth through useSession()

We did use the getServerSession(), its great but due to cold start on vercel, so it may take very long time without response when user lands on the page and the function has to run "cold boot" mode, but it has an advantage of not having an extra fetch to check for user session.

If we decide to move to client side, it would be faster on cold start as the page can be rendered instant, just the User component that needs auth would show skeleton loading, but it consumes more vercel bandwidth as it needs to fetch the user session through useSession() all the time.

Maybe my understanding is wrong, anyone have a preference?

Currently we use next-auth@4, nextjs@12, under vercel pro

**Comment by sangdth** [comment]
I'm trying to switch to Nextjs 13 with app router, and have similar questions. I think your understanding is valid, but I would love to see more opinions on this.

**Reply by sangdth** [reply]
Btw I can't remember where I read this, but it was something like, the most painful job is drawing things on browser (mean it's the slowest), so that's why I think Nextjs try to pull more and more stuffs into server side recently. So even with the cold start, we could prefer `getServerSession` when possible.
